src/hooks/useElementHeight.js

```diff
@@ -0,0 +1,16 @@
+import { useRef, useEffect, useState } from 'react';
+
+const useElementHeight = () => {
+  const ref = useRef(null);
+  const [height, setHeight] = useState(0);
+
+  useEffect(() => {
+    if (ref.current) {
+      setHeight(ref.current.offsetHeight);
+    }
+  }, []);
```

Comment:
❗️ 현재 `useEffect`에 의존성 배열이 비어있어 처음 마운트될 때 한 번만 실행됩니다.
이는 DOM 요소의 크기 변화를 감지하지 않으므로, 이후 요소 크기가 변경되더라도 상태가 업데이트되지 않습니다.
윈도우 크기 변경이나 리사이즈 이벤트를 감지해서 요소 크기를 업데이트해야 할 것 같습니다. `resize` 이벤트 리스너를 추가하면 화면 크기나 요소의 크기 변경을 처리할 수 있습니다.

❗️ `setHeight`는 `ref.current.offsetHeight`가 변경될 때마다 호출됩니다. `offsetHeight`는 화면 크기나 스타일에 따라 변할 수 있기 때문에, 매번 렌더링할 때마다 `setHeight`를 호출하는 것은 성능에 영향을 줄 수 있습니다. `useCallback`을 사용하면 `setHeight`와 같은 상태 업데이트 함수에서 상태가 변경될 때만 함수가 호출되도록 최적화할 수 있습니다.

```suggestion
    // setHeight 함수에 메모이제이션 적용
    const updateHeight = useCallback(() => {
      if (ref.current && ref.current.offsetHeight !== height) {
        setHeight(ref.current.offsetHeight);
      }
    }, [height]); // height 값이 바뀔 때만 함수가 변경되도록 함
  
    useEffect(() => {
      // 초기 높이 설정
      updateHeight();
  
      window.addEventListener('resize', updateHeight); // resize 이벤트 리스너 추가
  
      return () => {
        window.removeEventListener('resize', updateHeight); // cleanup
      };
    }, [updateHeight]); // updateHeight가 변경될 때마다 effect가 실행됨
  
    return [ref, height];
```

Reply:
오호 이럴떄 useCallback을 쓰면 되는군요...🤔 
안그래도 이 부분을 수정해야겠다고 생각하고 있었는데 덕분에 좋은 방법 배워갑니다!!